src/include/platform/internal/GenericPlatformManagerImpl_POSIX.cpp

```diff
@@ -132,18 +133,36 @@ CHIP_ERROR GenericPlatformManagerImpl_POSIX<ImplClass>::_StartChipTimer(int64_t
 template <class ImplClass>
 void GenericPlatformManagerImpl_POSIX<ImplClass>::_PostEvent(const ChipDeviceEvent * event)
 {
+    LockEventQueue();
     mChipEventQueue.push(*event); // Thread safe due to ChipStackLock taken by App thread
     SysOnEventSignal(this);       // Trigger wake select on CHIP thread
+    UnlockEventQueue();
 }
 
 template <class ImplClass>
 void GenericPlatformManagerImpl_POSIX<ImplClass>::ProcessDeviceEvents()
 {
+    LockEventQueue();
```

Comment:
This lock aquire is very large: this means every post event has to be locked while other events are processed.

Could we have it more granular by using event copies and only lock access/pop?

Reply:
I think we need to protect all access to mChipEventQueue, including "while (!mChipEventQueue.empty())", right?